recipes/stdgpu/all/conanfile.py

```diff
@@ -0,0 +1,211 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, rm, rmdir, replace_in_file
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.53.0"
+
+
+class StdgpuConan(ConanFile):
+    name = "stdgpu"
+    description = "Efficient STL-like Data Structures on the GPU"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://stotko.github.io/stdgpu/"
+    topics = ("cuda", "data-structures", "gpgpu", "gpu", "hip", "openmp", "rocm", "stl", "thrust")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "backend": ["cuda", "openmp", "hip"],
+        "setup_compiler_flags": [True, False],
+        "enable_contract_checks": [None, True, False],
+        "use_32_bit_index": [True, False],
+        "openmp": ["llvm", "system"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "backend": "openmp",
+        "setup_compiler_flags": False,
+        "enable_contract_checks": None,
+        "use_32_bit_index": True,
+        "openmp": "llvm",
+    }
+
+    @property
+    def _min_cppstd(self):
+        if self.version == "1.3.0":
+            return 14
+        else:
+            return 17
+
+    @property
+    def _compilers_min_versions(self):
+        if self.version == "1.3.0":
+            # Based on https://github.com/stotko/stdgpu/tree/1.3.0#building
+            return {
+                "gcc": "7",
+                "clang": "6",
+                "apple-clang": "6",
+                "msvc": "192",
+                "Visual Studio": "16",
+            }
+        else:
+            # > 1.3.0
+            # Based on https://github.com/stotko/stdgpu/tree/32e0517#building
+            return {
+                "gcc": "9",
+                "clang": "10",
+                "apple-clang": "12",
+                "msvc": "192",
+                "Visual Studio": "16",
+            }
+
+    def export_sources(self):
+        copy(self, "cmake/*", dst=self.export_sources_folder, src=self.recipe_folder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.options.backend != "openmp":
+            del self.options.openmp
```

Comment:
```suggestion
```

It will not work because options values are not evaluated yet in config_options, they will use default value always. You need to move it to configure instead. Only self.settings is evaluated here. More information here: https://docs.conan.io/2/reference/conanfile/methods/config_options.html